dali/operators/imgcodec/image_decoder.h

```diff
@@ -0,0 +1,796 @@
+      pinned_alloc_.struct_size = sizeof(nvimgcodecPinnedAllocator_t);
+      pinned_alloc_.struct_next = nullptr;
+      pinned_alloc_.pinned_malloc = static_dali_pinned_malloc;
+      pinned_alloc_.pinned_free = static_dali_pinned_free;
+      pinned_alloc_.pinned_ctx = mm::GetDefaultResource<mm::memory_kind::pinned>();
+      pinned_alloc_.pinned_mem_padding = spec.GetArgument<int64_t>("host_memory_padding");
+      pinned_alloc_ptr = &pinned_alloc_;
+    }
+
+    nvimgcodecInstanceCreateInfo_t instance_create_info{
+      NVIMGCODEC_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+      sizeof(nvimgcodecInstanceCreateInfo_t),
+      nullptr
+    };
+    instance_create_info.load_extension_modules = static_cast<int>(true);
+    instance_create_info.load_builtin_modules = static_cast<int>(true);
+    instance_create_info.extension_modules_path = nullptr;
+    instance_create_info.create_debug_messenger = static_cast<int>(true);
+    instance_create_info.debug_messenger_desc = nullptr;
+    instance_create_info.message_severity = verbosity2severity(2);
+    instance_create_info.message_category = NVIMGCODEC_DEBUG_MESSAGE_CATEGORY_ALL;
+
+    instance_ = NvImageCodecInstance::Create(&instance_create_info);
+
+    std::stringstream opts_ss;
+    size_t params_count = 0;
+    float hw_load = 0.65f;
+    for (auto pair : decoder_params_) {
+      if (pair.first == "use_fast_idct") {
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "libjpeg_turbo_decoder:fast_idct=" << std::any_cast<bool>(pair.second);
+      } else if (pair.first == "hybrid_huffman_threshold") {
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "nvjpeg_cuda_decoder:hybrid_huffman_threshold="
+                << std::any_cast<size_t>(pair.second);
+      } else if (pair.first == "hw_decoder_load") {
+        hw_load = std::any_cast<float>(pair.second);
+      } else if (pair.first == "jpeg_fancy_upsampling") {
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "nvjpeg_cuda_decoder:fancy_upsampling=" << std::any_cast<bool>(pair.second);
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "nvjpeg_hw_decoder:fancy_upsampling=" << std::any_cast<bool>(pair.second);
+      } else if (pair.first == "preallocate_width_hint") {
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "nvjpeg_hw_decoder:preallocate_width_hint="
+                << std::max(1, std::any_cast<int>(pair.second));
+      } else if (pair.first == "preallocate_height_hint") {
+        if (params_count)
+          opts_ss << " ";
+        opts_ss << "nvjpeg_hw_decoder:preallocate_height_hint="
+                << std::max(1, std::any_cast<int>(pair.second));
+      } else {
+        continue;
+      }
+      params_count++;
+    }
+
+    // TODO(janton): expose this or simply use good defaults
+    const char *e1 = std::getenv("NVIMGCODEC_NVJPEG_EXTRA_FLAGS");
```

Comment:
Where does this name come from? Can it be made more descriptive?

Reply:
This is no longer used. I'll remove